src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -354,49 +357,55 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
             long timeoutInMilliseconds,
             IntPtr context)
         {
-            // TODO: Handle <=0 values for the timeout
-            Task.Delay((int)timeoutInMilliseconds).ContinueWithStandard(context, (t, c) => CurlTimerElapsed((IntPtr)c));
+            if (timeoutInMilliseconds <= 0)
```

Comment:
I'm not understanding how this logic maps to what's specified in the docs.  From the docs it sounded like a positive value would set up a timer to fire, and then if -1 came in, that timer should be stopped, but the way this is set up, the previously set timer is still going to fire.  Seems like rather than using Task.Delay, it'd be better to just create a Timer and store it on the SafeCurlMultiHandle (or whatever state is associated with the provided context).  When you get one of these callbacks, you ensure the Timer is instantiated and then you Change it to have a dueTime of the value provided.  Timer's dueTime matches the described semantics, such that -1, 0, and a positive value should all do the right thing with regards to my understanding of the docs.  You can of course add some special-case optimizations if you want, such as if the Timer isn't instantiated and you get a 0 or a -1, don't bother creating it and do something else instead.